README.md

```diff
@@ -0,0 +1,182 @@
+# Firebase v1.0.0
+
+The Firebase library allows you to easily integrate with Firebase's realtime backend, which includes data storage, user authentication, static hosting, and more.
+
+**To add this library to your project, add `#require "Firebase.class.nut:1.0.0"` to the top of your agent code.**
+
+You can view the library’s source code on [GitHub](https://github.com/electricimp/Firebase/tree/v1.0.0).
+
+## Class Usage
+
+### Constructor: Firebase(*instance, [auth, domain, debug]*)
+
+The Firebase class must be instantiated with an instance name, and optionally an Auth Key, a custom Firebase domain, and a debug flag.
+
+| Parameter | Type   | Default            | Notes    |
+| --------- | ------ | ------------------ | -------- |
+| instance  | string | n/a                |          |
+| auth      | string | `null` - (no auth) | Optional |
+| domain    | string | firebaseio.com     | Optional |
+| debug     | bool   | `true`             | Optional - set to `false` to surpress error logging within the Firebase class |
+
+The domain and instance are used to construct the url requests are made against in the following was: https://{instance}.{domain}
+
+```squirrel
+#require "Firebase.class.nut:1.0.0"
+
+const FIREBASE_AUTH_KEY = "<-- Your Firebase Auth Key -->"
+
+firebase <- Firebase("myfirebase", FIREBASE_AUTH_KEY)
+```
+
+## Class Methods
+
+### on(*path, callback*)
```

Comment:
May want to mention that you must open a stream if you want to set up #on listeners